src/components/picture/_children/source.jsx

```diff
@@ -0,0 +1,44 @@
+import PropTypes from "prop-types";
+import formatSrcset from "../../../utils/format-srcset";
+
+const Source = ({ height, media, resizerOptions, resizerURL, src, width, ...rest }) => {
+	if (media && src) {
+		if (resizerOptions && resizerOptions.auth) {
+			return (
+				<source
+					{...rest}
+					height={height}
+					media={media}
+					srcSet={formatSrcset(resizerURL.concat(src), resizerOptions, width, height)}
+					width={width}
+				/>
+			);
+		}
+		return <source {...rest} media={media} srcSet={src} />;
+	}
+	return null;
+};
+
+Source.defaultProps = {
+	resizerOptions: {},
+	resizerURL: "",
+	width: null,
+	height: null,
+};
+
+Source.propTypes = {
+	/** The intrinsic height of the image in pixels */
+	height: PropTypes.number,
+	/** Media query used for the tag's `media` attribute */
+	media: PropTypes.string,
+	/** Options to pass into v2 resizer. */
+	resizerOptions: PropTypes.object,
+	/** The URL of the resizer service. Should have a trailing slash */
+	resizerURL: PropTypes.string,
+	/** The URL to an image to load and display. Should not have a leading slash */
+	src: PropTypes.string,
```

Comment:
@JackHowa : Wanted to ask about this. In the ticket, we have the following acceptance criteria.
> If no src, do not render the source tag

On line 6 I am setting a conditional for `(media && src)` and returning null if that's not the case. To be consistent with `Image`, should I instead make `src` a required field and only check for `(media)` in the conditional?

cc: @nschubach

Reply:
I'd make `media` and `src` required.  Then you wouldn't need to check for either of them.  Thinking though, we should probably properly test the thrown errors:

https://jestjs.io/docs/expect#tothrowerror

Reply:
If media and src are required - we shouldn't test the prop types throwing an error as that's testing the integration of anything library

Reply:
@matthewroach : Thank you. Do you agree that making the props required to stay consistent w/ `Image` would be preferable to making the `if (media && src) {` check?

Reply:
We should lean on prop types for required checks over using conditional logic within the block if we are just going to short circuit and exit on missing props like you have here

Reply:
Still need the conditional though to fulfill the AC, so I had him add that condition back in.  The PropTypes required only warns but still renders unfortunately.

Reply:
@matthewroach : Sounds great! I do have a check to not render `Source` if no `src` or `media`. That's in the ticket's acceptance criteria. 👍🏼